relay-quotas/src/redis.rs

```diff
@@ -292,20 +356,36 @@ impl<T: GlobalLimiter> RedisRateLimiter<T> {
                 // behave as well).
                 let retry_after = self.retry_after(REJECT_ALL_SECS);
                 rate_limits.add(RateLimit::from_quota(quota, *item_scoping, retry_after));
-            } else if let Some(quota) = RedisQuota::new(quota, item_scoping, timestamp) {
+            } else if let Some(mut quota) =
+                RedisQuota::new(quota, quantity, item_scoping, timestamp)
+            {
                 if quota.scope == QuotaScope::Global {
                     global_quotas.push(quota);
                 } else {
                     let key = quota.key();
+                    let redis_key = key.to_string();
+
+                    if let Some(cache) = &self.cache {
+                        let k = cache::Quota {
+                            limit: quota.limit(),
+                            key,
+                            expiry: UnixTimestamp::from_secs(quota.key_expiry()),
+                        };
+                        quota.quantity = match cache.check_quota(k, quantity) {
+                            cache::Action::Accept => continue,
+                            cache::Action::Check(quantity) => quantity,
+                        };
```

Comment:
IMO it's clearer to move the assignment into the branch.
```suggestion
                         match cache.check_quota(k, quantity) {
                            cache::Action::Accept => continue,
                            cache::Action::Check(quantity) => quota.quantity = quantity,
                        };
```

Reply:
I disagree, reading the block top to bottom with the assignment shows that the quantity is modified without having to read the branch.